app/views/admin/enterprises/form/_payment_methods.html.haml

```diff
@@ -10,9 +10,9 @@
         %th
     %tbody
       - @payment_methods.each do |payment_method|
-        %tr{ ng: { controller: 'paymentMethodsCtrl', init: "findPaymentMethodByID(#{payment_method.id})" } }
```

Comment:
I'm unsure just what angular was doing here. I thought it was deciding whether a checkbox should be checked or not. But it was not :shrug:
I have checked the behavior for payment methods in master and this branch and everything looks the same. I might have missed something though. Thanks!

Reply:
Yes, you're right: it decides wether or not the payment method should be checked or not. It's done in the next line, on the checkbox, via `'ng-model' => 'PaymentMethod.selected'` and `PaymentMethod` is set by `app/assets/javascripts/admin/payment_methods/controllers/payment_methods_controller.js.coffee` via `$scope.PaymentMethod = PaymentMethods.byID[id]`

```
%td= f.check_box :payment_method_ids, { multiple: true, 'ng-model' => 'PaymentMethod.selected' }, payment_method.id, nil
```

The default behavior (ie. without changing anything allows us to have some payment methods checked): 
<img width="953" alt="Capture d’écran 2022-08-05 à 09 35 16" src="https://user-images.githubusercontent.com/296452/183026832-c9cda605-c1d6-4a21-87c6-e22be7e331d4.png">

Otherwise, if I remove something that is relative to `PaymentMethod` (either the `ng-model` or the `ng-init`), the checkbox are unchecked: 
<img width="957" alt="Capture d’écran 2022-08-05 à 09 35 07" src="https://user-images.githubusercontent.com/296452/183027008-1fcc15c3-9a58-420a-a939-b65690345404.png">

So we need to take care of this one too... ;) 

Back in `In dev` then.

Reply:
(it seems to be done via `payment_method.id` and `payment_method_ids`).

Reply:
> (it seems to be done via `payment_method.id` and `payment_method_ids`).

Yup I thought so too but just couldn't figure out how, that's why I thought of mentioning it here thinking maybe I had broken something or maybe I just read the angular code wrong and it was doing something else.

Reply:
Yup it's using `@object.payment_method_ids` 
I used debugger to go into rails to see what it did, and it's using that to tick the check boxes. Curious that it's using `@object` and not `@enterprise` or `@payment_methods`. I thought it was using `@payment_methods` but `@payment_methods.payment_method_ids` didn't really make sense to me.